File: src/Entity/Visite.php
<?php

namespace App\Entity;

use App\Repository\VisiteRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: VisiteRepository::class)]
class Visite
{
   
   
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column()]
    private ?int $idvisite = null;
    /**
      *  @Assert\NotBlank(message="nom de visiteur est obligatoire")
     * @Assert\NotBlank(message="NB: nom bisiteur  doit etre Obligatoire et min caractere 2 et max 99")
     * @Assert\Length(
     *      min = 2,
     *      max = 100,
     *      minMessage = "doit etre >=4 ",
     *      maxMessage = "doit etre <=100" )
     * @Groups("visite")
     */


    #[ORM\Column(length: 255)]
    private ?string $nomvisiteur = null;
    /**
      *  @Assert\NotBlank(message="message de patient est obligatoire")
     * @Assert\NotBlank(message="NB: nom patient  doit etre Obligatoire et min caractere 2 et max 99")
     * @Assert\Length(
     *      min = 2,
     *      max = 100,
     *      minMessage = "doit etre >=4 ",
     *      maxMessage = "doit etre <=100" )
     *  @Groups("visite")
     */


    #[ORM\Column(length: 255)]
    private ?string $nompatient = null;
    /**
      *  @Assert\NotBlank(message="date de visite est obligatoire")
      * @Groups("visite")
      */

    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    private ?\DateTimeInterface $date = null;
    /**
      *  @Assert\NotBlank(message="numero de visiteur est obligatoire")
     * @Assert\NotBlank(message="NB: numero de visiteur doit etre Obligatoire et min 8 et max 12")
     * @Assert\Length(
     *      min = 8,
     *      max =15,
     *      minMessage = "doit etre >=8 ",
     *      maxMessage = "doit etre <=13" )
     *  @Groups("visite")
     */

    #[ORM\Column]
    private ?int $numerotele = null;
      /**
      *  @Assert\NotBlank(message="message de visiteur est obligatoire")
      * @Groups("visite")
     * 
     */

    #[ORM\Column(type: Types::TEXT)]
    private ?string $message = null;

    #[ORM\ManyToOne(inversedBy: 'visites')]
    private ?Patient $patient = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $image = null;

    public function getIdvisite(): ?int
    {
        return $this->idvisite;
    }

   
   
    public function setIdvisite(int $idvisite): self
    {
        $this->getidvisite = $idvisite;

        return $this;
    }

    public function getNomvisiteur(): ?string
    {
        return $this->nomvisiteur;
    }

    public function setNomvisiteur(string $nomvisiteur): self
    {
        $this->nomvisiteur = $nomvisiteur;

        return $this;
    }

    public function getNompatient(): ?string
    {
        return $this->nompatient;
    }

    public function setNompatient(string $nompatient): self
    {
        $this->nompatient = $nompatient;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): self
    {
        $this->date = $date;

        return $this;
    }

    public function getNumerotele(): ?int
    {
        return $this->numerotele;
    }

    public function setNumerotele(int $numerotele): self
    {
        $this->numerotele = $numerotele;

        return $this;
    }

    public function getMessage(): ?string
    {
        return $this->message;
    }

    public function setMessage(string $message): self
    {
        $this->message = $message;

        return $this;
    }

    public function getPatient(): ?Patient
    {
        return $this->patient;
    }

    public function setPatient(?Patient $patient): self
    {
        $this->patient = $patient;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): self
    {
        $this->image = $image;

        return $this;
    }
}

File: entities/Visite.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package edu.connexion3a41.entities;

import java.sql.Date;

/**
 *
 * @author tha3leb
 */
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


/**
 *
 * @author tha3leb
 */
public class Visite {
     private int idvisite;
    private String nomvisiteur ,nompatient;
     private Date date;
     
    private String numerotele;
    private String message;
       private int patient_id;

    public Visite(int idvisite, String nomvisiteur, String nompatient, Date date, String numerotele, String message, int patient_id) {
        this.idvisite = idvisite;
        this.nomvisiteur = nomvisiteur;
        this.nompatient = nompatient;
        this.date = date;
        this.numerotele = numerotele;
        this.message = message;
        this.patient_id = patient_id;
    }

    public Visite(String nomvisiteur, String nompatient, Date date, String numerotele, String message, int patient_id) {
        this.nomvisiteur = nomvisiteur;
        this.nompatient = nompatient;
        this.date = date;
        this.numerotele = numerotele;
        this.message = message;
        this.patient_id = patient_id;
    }

    public Visite(int idvisite) {
        this.idvisite = idvisite;
    }


    public int getIdvisite() {
        return idvisite;
    }

    public void setIdvisite(int idvisite) {
        this.idvisite = idvisite;
    }

    public String getNomvisiteur() {
        return nomvisiteur;
    }

    public void setNomvisiteur(String nomvisiteur) {
        this.nomvisiteur = nomvisiteur;
    }

    public String getNompatient() {
        return nompatient;
    }

    public void setNompatient(String nompatient) {
        this.nompatient = nompatient;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getNumerotele() {
        return numerotele;
    }

    public void setNumerotele(String numerotele) {
        this.numerotele = numerotele;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public int getPatient_id() {
        return patient_id;
    }

    public void setPatient_id(int patient_id) {
        this.patient_id = patient_id;
    }

    @Override
    public String toString() {
        return "Visite{" + "idvisite=" + idvisite + ", nomvisiteur=" + nomvisiteur + ", nompatient=" + nompatient + ", date=" + date + ", numerotele=" + numerotele + ", message=" + message + ", patient_id=" + patient_id + '}';
    }

  
           
}

